### ISSUE TITLE
Forward OAuth Identity does not work for queries

### ISSUE THREAD

**Issue by anemyte** [grafana/opensearch-datasource#487]
**What happened**:
This plugin does not include `Authorization` header when it makes queries to an opensearch server despite `Forward OAuth Identity` being toggled on. This is indicated by this error:
```
Post "https://opensearch.example.com:9200/_msearch?max_concurrent_shard_requests=5": downstream error: 401 Unauthorized
```

**What you expected to happen**:
I expect the plugin to actually `Forward OAuth Identity` as it was configured to do. I suppose it is done by supplementing query requests with current user's access_token via Authorization HTTP header.

**How to reproduce it (as minimally and precisely as possible)**:
1. Setup an Opensearch server
2. Login to your Grafana instance using an OpenID-Connect identity provider
3. Configure this plugin to use the opensearch server from p1 and toggle `Forward OAuth Identity` on
4. Use `tcpdump` to capture Grafana requests
5. Try executing a query via explore or a dashboard panel.
6. Use Wireshark or an alternative to analyze packet dump from p.4 and notice missing Authorization header in query requests.

**Anything else we need to know?**:
The request to get Opensearch version from plugin configuration page does actually come with the user's identity. Here is how this request looks in the packet capture:
![Image](https://github.com/user-attachments/assets/95c0524b-a782-49a7-b91f-37e4f30d8b6f)

And this is how a query request looks like:
![Image](https://github.com/user-attachments/assets/af294979-4a07-4ece-adbc-03243b17836b)

As you can see, the second request comes without the header and thus it gets the 401 error from opensearch.

Another thing that I want to point out is that the Elasticsearch datasource plugin configured to use the same opensearch server in the same environment does actually work, I can see query results and no error. Here's how the Elasticsearch plugin performs the request:
![Image](https://github.com/user-attachments/assets/ede56343-0d87-498a-86f4-0909ddc01c80)

**Environment**:
- Grafana version: v11.3.0 (d9455ff7db)
- OpenSearch version: 2.16.0
- Plugin version: 2.20.0, 2.21.0 (tested both)
- Identity Provider: Keycloak v25.0.2

**Comment by idastambuk** [comment]
Hi @anemyte can you try updating your plugin to the latest version, 2.21.1 and see if that changes if the headers are being passed in get_version and queryData? Thanks!

**Comment by anemyte** [comment]
@idastambuk Well, I don't see any changes with 2.21.1. "get_version" still passes the header and works, while querying still doesn't. Here's the query request once again: 

![Image](https://github.com/user-attachments/assets/b4b7cdde-2973-4d11-8bb2-3c055a79e748)

**Comment by anemyte** [comment]
@idastambuk Do you need any help in setting up the environment to work on this? I understand it can be a little too hard to launch and configure all the services. I could write a docker-compose file for you to spin up the environment locally (if you have docker installed, of course).

**Comment by iwysiu** [comment]
Hi @anemyte , I'm currently taking a look at this, but to help me figure out what path to investigate, do you have the feature flag `openSearchBackendFlowEnabled` set to true? If you're on hosted grafana it should be on by default. (For context we've been migrating Opensearch to backend queries, and are planning on deprecating the frontend path). If you're not sure, if you send me a screenshot of network information for the frontend request and I can figure it out.

**Comment by iwysiu** [comment]
Also, a docker file to spin up the test environment would be very helpful, thanks!

**Comment by anemyte** [comment]
@iwysiu I have a self-hosted Grafana and I'm pretty sure I did not touch that feature toggle, so it should be on default (not really know how to check for sure). There is an option in datasource settings called "Access" and it is set to "Server", I guess this must be the "backend flow".

Anyways, here's the promised docker compose: https://github.com/anemyte/opensearch-plugin-test-env
It should be relatively simple to launch (just one command) but let me know if there are problems with it.

**Comment by idastambuk** [comment]
Hi @anemyte we just released [v2.21.5](https://github.com/grafana/opensearch-datasource/releases/tag/v2.21.5)  that should take care of the headers. Let us know if this works for you. 
Thanks a lot for the repro repository, it was invaluable!
